From e80f88776a350a3e6c68767ed5a9776d8fe46803 Mon Sep 17 00:00:00 2001
From: Khai Wen Ng <khai.wen.ng@intel.com>
Date: Thu, 11 Sep 2025 09:31:53 +0800
Subject: [PATCH 52/61] media: i2c: maxim-serdes: max_ser: retrieve MUX adapter
 ID for serializer

Retrieve and use the ID for child MUX Adapter if ID is found in FWnode.

Signed-off-by: Khai Wen Ng <khai.wen.ng@intel.com>
---
 drivers/media/i2c/maxim-serdes/max_ser.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/maxim-serdes/max_ser.c b/drivers/media/i2c/maxim-serdes/max_ser.c
index 00e2e3417b49e..583f3c2a24243 100644
--- a/drivers/media/i2c/maxim-serdes/max_ser.c
+++ b/drivers/media/i2c/maxim-serdes/max_ser.c
@@ -414,13 +414,22 @@ static void max_ser_i2c_mux_deinit(struct max_ser_priv *priv)
 
 static int max_ser_i2c_mux_init(struct max_ser_priv *priv)
 {
+	u32 adapter;
+
 	priv->mux = i2c_mux_alloc(priv->client->adapter, &priv->client->dev,
 				  1, 0, I2C_MUX_LOCKED | I2C_MUX_GATE,
 				  max_ser_i2c_mux_select, NULL);
 	if (!priv->mux)
 		return -ENOMEM;
 
-	return i2c_mux_add_adapter(priv->mux, 0, 0);
+	/* serializer only have 1 input */
+	if (fwnode_property_present(dev_fwnode(priv->dev), "mux-adapter"))
+		fwnode_property_read_u32(dev_fwnode(priv->dev), "mux-adapter", &adapter);
+
+	if (adapter < 0)
+		adapter = 0;
+
+	return i2c_mux_add_adapter(priv->mux, adapter, 0);
 }
 
 static int max_ser_i2c_adapter_init(struct max_ser_priv *priv)
-- 
2.50.1

